src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -86,6 +90,21 @@ public void Dispose()
             this.textWriter = null;
         }
 
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+        public ValueTask DisposeAsync()
+        {
+            return DisposeAsyncCore();
+        }
+
+        private async ValueTask DisposeAsyncCore()

```

Comment:
Is it necessary to create a `DisposeAsyncCore` method if this class is sealed?

From the docs it sounds like the `public DisposeAsync` + `protected virtual DisposeAsyncCore` pattern is suitable for non-sealed classes.

Not that it hurts in this case, just curious.

Reply:
Changed this to a local function within `DisposeAsync`